packages/react-hooks/src/hooks/tests/timeout.test.tsx

```diff
@@ -18,38 +16,84 @@ describe('useTimeout', () => {
     timer.restore();
   });
 
-  it('does not call the handler immediately', () => {
-    const spy = jest.fn();
-    const delay = 1000;
+  it('does not call the callback immediately', () => {
+    const {callback} = setup({delay: ONE_SECOND});
 
-    mount(<MockComponent delay={delay} spy={spy} />);
+    expect(callback).not.toHaveBeenCalled();
+  });
 
-    expect(spy).not.toHaveBeenCalled();
+  it('calls the callback exactly once after the delay', () => {
+    const {callback} = setup({delay: ONE_SECOND});
+
+    timer.runAllTimers();
+    expect(callback).toHaveBeenCalledTimes(1);
   });
 
-  it('calls the handler after the delay', () => {
-    const spy = jest.fn();
-    const delay = 1000;
+  it('updates the callback when it changes', () => {
+    const {callback, wrapper} = setup({delay: ONE_SECOND});
 
-    mount(<MockComponent delay={delay} spy={spy} />);
+    const newCallback = jest.fn();
+    wrapper.setProps({callback: newCallback});
 
     timer.runAllTimers();
+    expect(newCallback).toHaveBeenCalled();
+    expect(callback).not.toHaveBeenCalled();
+  });
+
+  it('sets no timeout if the delay provided is null', () => {
+    const {callback} = setup({delay: null});
+
+    timer.runTimersToTime(ONE_SECOND);
 
-    expect(spy).toHaveBeenCalled();
+    expect(callback).not.toHaveBeenCalled();
   });
 
-  it('unsubscribes timeouts when the passed in callback changes', () => {
-    const spy = jest.fn();
-    const delay = 1000;
-    const newSpy = jest.fn();
+  it('clears existing timeout if the new delay is null', () => {
+    const {callback, wrapper} = setup({delay: ONE_SECOND});
 
-    const wrapper = mount(<MockComponent delay={delay} spy={spy} />);
+    wrapper.setProps({delay: null});
 
-    wrapper.setProps({spy: newSpy});
+    timer.runTimersToTime(ONE_SECOND);
+    expect(callback).not.toHaveBeenCalled();
+  });
 
-    timer.runAllTimers();
+  it('continues to respect the delay after the callback changes', () => {
+    const {callback, wrapper} = setup({delay: ONE_SECOND});
 
-    expect(spy).not.toHaveBeenCalled();
-    expect(newSpy).toHaveBeenCalled();
+    timer.runTimersToTime(HALF_A_SECOND);
+    expect(callback).not.toHaveBeenCalled();
+
+    const newCallback = jest.fn();
+    wrapper.setProps({callback: newCallback});
+
+    timer.runTimersToTime(HALF_A_SECOND);
+    expect(newCallback).toHaveBeenCalled();
+    expect(callback).not.toHaveBeenCalled();
+  });
```

Comment:
I'm not sure if this should be a breaking change or not. Technically, it does break the previous behaviour, though it would be weird behaviour to depend on, so I'm leaning towards `breaking`, but I want to confirm with reviewers.

### Previously

Changing the `callback` prop causes the timeout to be reset, effectively restarting the timer. If the timeout has already fired, a new timeout will be set, and the new callback will be run (in addition to the one that already ran).

### Now

Changing the `callback` does nothing. When the timeout fires, it will run whatever the current timeout is. If the timeout has already fired, nothing will happen, unless the `delay` is also changed.

Reply:
I kinda see this as a bugfix tbh

Reply:
👌 I've updated the Changelog to include two entries.
- Added support for `null` `delay`
- Fixed changing callback resetting the timeout